nexus/src/app/sagas/vpc_subnet_create.rs

```diff
@@ -0,0 +1,605 @@
+        + svsc_create_route
+        - svsc_create_route_undo
+    }
+    VPC_SUBNET_CREATE_LINK_CUSTOM -> "output" {
+        + svsc_link_custom
+        - svsc_link_custom_undo
+    }
+    VPC_NOTIFY_RPW -> "notified" {
+        + svsc_notify_rpw
+    }
+}
+
+// vpc subnet create saga: definition
+
+#[derive(Debug)]
+pub(crate) struct SagaVpcSubnetCreate;
+impl NexusSaga for SagaVpcSubnetCreate {
+    const NAME: &'static str = "vpc-subnet-create";
+    type Params = Params;
+
+    fn register_actions(registry: &mut ActionRegistry) {
+        vpc_subnet_create_register_actions(registry);
+    }
+
+    fn make_saga_dag(
+        _params: &Self::Params,
+        mut builder: steno::DagBuilder,
+    ) -> Result<steno::Dag, super::SagaInitError> {
+        builder.append(Node::action(
+            "subnet_id",
+            "GenerateVpcSubnetId",
+            ACTION_GENERATE_ID.as_ref(),
+        ));
+        builder.append(Node::action(
+            "route_id",
+            "GenerateRouteId",
+            ACTION_GENERATE_ID.as_ref(),
+        ));
+
+        builder.append(vpc_subnet_create_subnet_action());
+        builder.append(vpc_subnet_create_sys_route_action());
+        builder.append(vpc_subnet_create_link_custom_action());
+        builder.append(vpc_notify_rpw_action());
+
+        Ok(builder.build()?)
+    }
+}
+
+// vpc subnet create saga: action implementations
+
+async fn svsc_create_subnet(
+    sagactx: NexusActionContext,
+) -> Result<(authz::VpcSubnet, db::model::VpcSubnet), ActionError> {
+    let osagactx = sagactx.user_data();
+    let params = sagactx.saga_params::<Params>()?;
+    let opctx = crate::context::op_context_for_saga_action(
+        &sagactx,
+        &params.serialized_authn,
+    );
+    let log = sagactx.user_data().log();
+
+    let subnet_id = sagactx.lookup::<Uuid>("subnet_id")?;
+    let vpc_id = params.authz_vpc.id();
+
+    let num_retries = params.ipv6_blocks.len();
+    let retryable = num_retries > 1;
+
+    let mut result = None;
+    for ipv6_block in params.ipv6_blocks {
+        let subnet = db::model::VpcSubnet::new(
+            subnet_id,
+            vpc_id,
+            params.subnet_create.identity.clone(),
+            params.subnet_create.ipv4_block,
+            ipv6_block,
+        );
+
+        result = Some(
+            osagactx
+                .datastore()
+                .vpc_create_subnet(&opctx, &params.authz_vpc, subnet)
+                .await,
+        );
+
+        if matches!(result, Some(Ok(_))) {
+            break;
+        }
+
+        match &result {
+            Some(Ok(_)) => break,
+            // Allow NUM_RETRIES retries, after the first attempt.
+            //
+            // Note that we only catch IPv6 overlaps. The client
+            // always specifies the IPv4 range, so we fail the
+            // request if that overlaps with an existing range.
+            Some(Err(InsertVpcSubnetError::OverlappingIpRange(IpNet::V6(
```

Comment:
Most comments here have been transcribed from the original `vpc_create_subnet` method. This should be replicating the original behaviour (3 random attempts, OR fail on explicit collision) but in a saga-friendly/idempotent way.